docs/source/perf/assume_pure.md

```diff
@@ -0,0 +1,131 @@
+# Use `@assume_pure` to speed up lazy tensor tracing
+
+This document guides how to use `torch_xla.experimental.assume_pure` to eliminate
+lazy tensor tracing overhead. See [this blog post][lazy-tensor] for a primer on
+how lazy tensor tracing (operation recording) works.
+
+## Background and motivation
+
+PyTorch/XLA's lazy tensor tracing ensures correct execution by recording an
+operation graph (lazy tensor IR) when running PyTorch operations. However, for
+complex models, this tracing overhead can exceed the execution time of the
+graph, leading to performance bottlenecks. When training a model, the layers in
+the model must be re-traced on every training step. That's because in the limit,
```

Comment:
That's because there's no guarantee that the layers will do the same thing in different training steps.